node/impl/full/eth.go

```diff
@@ -1276,8 +1276,14 @@ func (e *EthEventHandler) EthGetLogs(ctx context.Context, filterSpec *ethtypes.E
 	if pf.tipsetCid == cid.Undef {
 		maxHeight := pf.maxHeight
 		if maxHeight == -1 {
-			maxHeight = e.Chain.GetHeaviestTipSet().Height()
+			// we can't use the heaviest tipset here as that could block the user on block execution
+			maxHeight = e.Chain.GetHeaviestTipSet().Height() - 1
 		}
+
+		if maxHeight < 0 {
+			return nil, xerrors.Errorf("maxHeight requested is less than 0")
+		}
+
 		if maxHeight > e.Chain.GetHeaviestTipSet().Height() {
```

Comment:
this is an interesting case; if they request this then they'll block for <=30s, maybe we don't want that? :shrug:

Reply:
I agree. I've fixed this. Looks like `eth_getBlockByNumber` also returns the head -1 tipset for "latest"

```
func getTipsetByBlockNumber(ctx context.Context, chain *store.ChainStore, blkParam string, strict bool) (*types.TipSet, error) {
	if blkParam == "earliest" {
		return nil, fmt.Errorf("block param \"earliest\" is not supported")
	}

	head := chain.GetHeaviestTipSet()
	switch blkParam {
	case "pending":
		return head, nil
	case "latest":
		parent, err := chain.GetTipSetFromKey(ctx, head.Parents())
		if err != nil {
			return nil, fmt.Errorf("cannot get parent tipset")
		}
		return parent, nil
```